tools/src/main/java/org/apache/kafka/trogdor/agent/WorkerManager.java

```diff
@@ -345,6 +352,7 @@ public void createWorker(long workerId, String taskId, TaskSpec spec) throws Thr
                     "worker.start() exception: " + Utils.stackTrace(e), true));
             }
             stateChangeExecutor.submit(new FinishCreatingWorker(worker));
+            return worker.doneFuture;
```

Comment:
I don't think this is accurate.  `HandleWorkerHalting` is supposed to get the worker to `DONE`.  For example, if the problem is that the worker halted during startup, it just calls `transitionToDone` directly.  If the worker halted after startup, then it will have to go through other states like `CANCELLING` first, but it will arrive at `DONE` eventually.